tests/integration/gateway/verify_cgw_traffic_test.go

```diff
@@ -0,0 +1,156 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/shell"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	vsTemplate = `
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: %s
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - %s
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: %s
+`
+	gwTemplate = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: %s
+spec:
+  selector:
+    istio: %s
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+`
+)
+
+// TestAccessAppViaCustomGateway tests access to an aplication using a custom gateway
+func TestAccessAppViaCustomGateway(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("traffic.ingress.gateway").
+		Run(func(ctx framework.TestContext) {
+
+			// Unable to find the ingress for the custom gateway install via the framework so retrieve URL and
```

Comment:
should we just fix the framework?

Reply:
Agree that we should, if the framework is broken. It might just be what I was doing. I was hoping that maybe would tell be able to tell me what I might have been doing wrong, although I guess I don't have the code here that I thought should have worked.